Dockerfile

```
@@ -1,64 +1,74 @@
-# Base stage for shared dependencies
-FROM node:20-alpine AS base
-RUN apk add --no-cache python3 make g++
-RUN npm install -g pnpm
+## Base layer: install shared dependencies once (leverages Docker layer caching)
+FROM node:24-alpine AS base
+RUN apk add --no-cache python3 make g++ \
```

Hadolint (2.14.0)

[warning] Pin versions in npm. Instead of `npm install <package>` use `npm install <package>@<version>`

(DL3016)